src/vs/editor/contrib/stickyScroll/browser/stickyScrollWidget.ts

```diff
@@ -104,8 +132,132 @@ export class StickyScrollWidget extends Disposable implements IOverlayWidget {
 		}
 	}
 
-	private _renderChildNode(index: number, line: number): HTMLDivElement {
+	private _renderCodeLensLine(lineNumber: number) {
+		this._commands.clear();
+
+		const codeLensItems = this._groupCodeLensModel(this._codeLensModel);
+		if (!codeLensItems?.length) {
+			return;
+		}
+
+		const lineCodeLensItems = codeLensItems.find(cl => cl[0].symbol?.range?.startLineNumber === lineNumber);
+		if (!lineCodeLensItems?.length) {
+			return;
+		}
 
+		const child = document.createElement('div');
+		const layoutInfo = this._editor.getLayoutInfo();
+		const width = layoutInfo.width - layoutInfo.minimap.minimapCanvasOuterWidth - layoutInfo.verticalScrollbarWidth;
+		child.className = 'sticky-line-root';
+		child.setAttribute('role', 'listitem');
+		child.tabIndex = 0;
+		child.style.cssText = `width: ${width}px; z-index: 0;`;
+		const codeLensCont = document.createElement('div');
+		codeLensCont.className = 'codelens-decoration';
+		codeLensCont.style.cssText = `position: relative; left: ${layoutInfo.contentLeft}px;`;
+		child.appendChild(codeLensCont);
+
+		// Convert codelens startColumn to space and use renderViewLine to render it
```

Comment:
@aiday-mar Is there a better way to compute the Code Lens's marginLeft?

Reply:
As now I use renderViewLine to render spaces according to the startColumn, which I think is not a lightweight solution,

Reply:
@VSCodeTriageBot Any one can handle this pull request?

Reply:
@harbin1053020115 the team are in the middle of endgame, so I don't think this PR will get looked at until afterwards.

Reply:
> @harbin1053020115 the team are in the middle of endgame, so I don't think this PR will get looked at until afterwards.

@gjsjohnmurray You mean May 2023 Endgame(https://github.com/microsoft/vscode/issues/183504)？

Reply:
Yes, #183504

Reply:
> Yes, #183504

Got it~